src/Functions/URL/domain.h

```diff
@@ -20,6 +20,115 @@ inline std::string_view checkAndReturnHost(const Pos & pos, const Pos & dot_pos,
     return std::string_view(start_of_host, pos - start_of_host);
 }
 
+/// Extracts host from given url (RPC).
+///
+/// @return empty string view if the host is not valid (i.e. it does not have dot, or there no symbol after dot).
+inline std::string_view getURLHostRFC(const char * data, size_t size)
```

Comment:
New functions are still non RFC conformant, here are some issues with it:

- maximum schema length is 16 (this is an optimization likely, because it should work for both, URL and domain)
- restricted symbols are wrong, they should be like this:

```

      unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
      pct-encoded = "%" HEXDIG HEXDIG
      sub-delims  = "!" / "$" / "&" / "'" / "(" / ")"
                  / "*" / "+" / "," / ";" / "="

      gen-delims  = ":" / "/" / "?" / "#" / "[" / "]" / "@"
      reserved    = gen-delims / sub-delims

      userinfo    = *( unreserved / pct-encoded / sub-delims / ":" )
      host        = IP-literal / IPv4address / reg-name
      reg-name    = *( unreserved / pct-encoded / sub-delims )
```

While here lots of this symbols are prohibited (`~`, `%`, ...)

According to https://www.rfc-editor.org/rfc/rfc3986

Reply:
And while it is more or less OK to ignore lots of symbols in host name (since unlikely you will be able to register such domain, though you can always server wildcard domains from your server directly, and in this case you depends on what browsers detects ad domain), it could be a problem to not allow such symbols in userinfo

Reply:
Here are some changes - https://github.com/ClickHouse/ClickHouse/pull/46841